torch_xla/utils/checkpoint_tagger.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import json
+
+class CheckpointTagger(object):
+
+  def __init__(self, remover=None):
+    self._tags = dict()
+    self._refcount = collections.defaultdict(int)
+    remover = (lambda x: None) if remover is None else remover
+    assert callable(remover)
+    self._remover = remover
+
+  def tag(self, name, path):
+    self._refcount[path] += 1
+    old_path = self._tags.get(name)
+    if old_path is not None:
+      self._refcount[old_path] -= 1
+      if self._refcount[old_path] == 0:
+        self._refcount.pop(old_path)
+        self._remover(old_path)
+    self._tags[name] = path
+
+  def get_tag(self, tag):
```

Comment:
```Python
def get_tags(self):
  return self._tags
```

Reply:
oh I thought you wanted an api to get the `last` checkpoint (or any other tag). That we're going to use when loading. Why do we need this (to get all the tags)?

Reply:
No, an API to just expose the whole tags dictionary.

Reply:
ok, I'll add that. but I need to add this api too, are you objecting to this one? I'm thinking of making this `def __getitem__(self, tag): return self._tags.get(tag)`, wdyt?

Reply:
Why do you need that API?
Again:
```Python
def get_tags(self):
  return self._tags
```

Your API becomes:
```Python
ch.get_tags().get('foo', None)
```

Reply:
imho it's much nicer if I can just get the last checkpoint from this instead of getting all the tags. That's how this will be used the most, don't you think?

* I want to resume from the last checkpoint, I only need the last checkpoint
* I want to predict using the best checkpoitn, I only need the best one.

etc

Reply:
You are not getting any copies of everything, just a reference 😉 
```Python
@property
def tags(self):
  return self._tags
```
```Python
ch.tags.get('foo')
```
But also:
```Python
for k, v in ch.tags.items():
  # Look Ma! I can iterate :)
```

Reply:
I know, it just makes the code that uses this tool a bit uglier imho, but not a big deal, so I deleted that part.